packages/jira-adapter/src/filters/fields/context_options.ts

```diff
@@ -108,17 +109,31 @@ const updateContextOptions = async ({
   baseUrl,
   contextChange,
 }: UpdateContextOptionsParams): Promise<void> => {
-  if (addedOptions.length !== 0) {
-    const addedOptionsChunks = _.chunk(addedOptions, OPTIONS_MAXIMUM_BATCH_SIZE)
+  if (removedOptions.length !== 0) {
+    await Promise.all(removedOptions.map(async (option: Value) => {
+      await client.delete({
+        url: `${baseUrl}/${option.id}`,
+      })
+    }))
+  }
 
+  if (addedOptions.length !== 0) {
+    const optionLengthBefore = isModificationChange(contextChange)
+      ? getOptionsFromContext(contextChange.data.before).length : 0
+    // eslint-disable-next-line no-nested-ternary
+    const sliceIndex = optionLengthBefore > OPTIONS_MAXIMUM_SIZE
+      ? 0 : optionLengthBefore + addedOptions.length > OPTIONS_MAXIMUM_SIZE
+        ? OPTIONS_MAXIMUM_SIZE - optionLengthBefore : addedOptions.length
+    const firstOptions = addedOptions.slice(0, sliceIndex)
+    const secondOptions = addedOptions.slice(sliceIndex, addedOptions.length)
```

Comment:
nit: variable names:
```suggestion
    const privateApiOptions = addedOptions.slice(0, sliceIndex)
    const publicApiOptions = addedOptions.slice(sliceIndex, addedOptions.length)
```